### PR TITLE
feat: check if operations being added are already stored

### PR DESCRIPTION
**Pull request by acaldas**
I fixed a wrong configuration on my IDE where Prettier was not being run. This caused some reformats on this PR.

When adding new operations, gets the document from cache or storage, and if all the operations being added are already known then skips all work and returns SUCCESS.

This makes `PushUpdates` from Connect a lot faster and reduces their load on Switchboard.

### CODE REVIEW

File: src/storage/prisma.ts [hunk 1]
```diff
@@ -42,6 +47,7 @@ function storageToOperation(
     }
 ): Operation {
     const operation: Operation = {
+        id: op.opId || undefined,
```
Comment (acaldas) [inline src/storage/prisma.ts:50]: We were not loading the operation id from prisma

File: src/server/index.ts [hunk 2]
```diff
@@ -1273,33 +1479,92 @@ export class DocumentDriveServer extends BaseDocumentDriveServer {
...
-                }
-            });
-            const unsubscribeError = this.queueManager.on('jobFailed', (job, error) => {
-                if (job.jobId === jobId) {
-                    unsubscribe();
-                    unsubscribeError();
-                    reject(error);
-                }
-            });
+        // if operations are already stored then returns cached document
```
Comment (acaldas) [inline src/server/index.ts:1528]: Code change here

File: src/server/index.ts [hunk 3]
```diff
@@ -1273,33 +1479,92 @@ export class DocumentDriveServer extends BaseDocumentDriveServer {
...
-    queueDriveOperation(drive: string, operation: Operation<DocumentDriveAction | BaseAction>, forceSync = true): Promise<IOperationResult<DocumentDriveDocument>> {
+    queueDriveOperation(
+        drive: string,
+        operation: Operation<DocumentDriveAction | BaseAction>,
+        forceSync = true
+    ): Promise<IOperationResult<DocumentDriveDocument>> {
         return this.queueDriveOperations(drive, [operation], forceSync);
     }
 
+    private async resultIfExistingDriveOperations(
```
Comment (acaldas) [inline src/server/index.ts:1490]: Existing operations check here

File: src/server/index.ts [hunk 4]
```diff
@@ -1012,93 +1106,196 @@ export class DocumentDriveServer extends BaseDocumentDriveServer {
...
+        }
+    }
 
+    async queueOperations(
+        drive: string,
+        id: string,
+        operations: Operation[],
+        forceSync = true
+    ) {
+        // if operations are already stored then returns cached document
```
Comment (acaldas) [inline src/server/index.ts:1158]: Code change here

File: src/server/index.ts [hunk 5]
```diff
@@ -1012,93 +1106,196 @@ export class DocumentDriveServer extends BaseDocumentDriveServer {
...
+        drive: string,
+        id: string,
+        operation: Operation,
+        forceSync = true
+    ): Promise<IOperationResult> {
         return this.queueOperations(drive, id, [operation], forceSync);
     }
 
-    async queueOperations(drive: string,
+    private async resultIfExistingOperations(
```
Comment (acaldas) [inline src/server/index.ts:1118]: Existing operations check here

**Review by froid1911 — APPROVED** [review]

**Pull request merged**